src/components/ExperienceDetail/ReportForm/ReportForm.js

```diff
@@ -33,122 +33,126 @@ export const reasonCategoryOptions = [
   },
 ];
 
-class ReportForm extends PureComponent {
-  constructor(props) {
-    super(props);
+const ReportForm = ({ close, onApiError, onSuccess, id }) => {
+  const dispatch = useDispatch();
+  const [isLoggedIn, login] = useLogin();
+  const [formState, setFormState] = useState({
```

Comment:
想說把兩個值拆成兩個state會不會更好維護一點呢

Reply:
本來是想拆開兩個值，但發現原本的邏輯把這兩個參數綁在一起做 validation

```js
validReasomForm(this.state);
handleToApiParams(this.state),
```

Reply:
我想像是可以這樣做
```suggestion
validReasomForm({ reason, reasonCategory });
handleToApiParams({ reason, reasonCategory }),
```

維護時方開維護，驗證時只是組合